### ISSUE TITLE
Yazi uses incorrect prefixes with the file/directory sizes.

### ISSUE THREAD

**Issue by gedgon** [sxyazi/yazi#579]
### What system are you running Yazi on?

Linux Wayland

### What terminal are you running Yazi in?

foot 1.16.2

### Yazi version

yazi 0.2.2 (ba0c860 2024-01-26)

### Did you try the latest main branch to see if the problem has already been fixed?

Tried, but the problem is still present

### Describe the bug

Yazi counts in IEC (power of two), but displays JEDEC ~~(power of ten)~~ prefixes (KB, MB, GB, TB).

### Expected Behavior

Yazi should display the correct prefixes, which is in this case: KiB, MiB, GiB, TiB, etc.

**Comment by sxyazi** [comment]
Thanks for the reporting!

How about we use `K`, `M`, `G`, etc. instead of the current `KB`, `MB`, `GB`, which will match the behavior of `ls -lah`, and it's also shorter, like this:

![1](https://github.com/sxyazi/yazi/assets/17523360/4766d12a-7dd3-447e-a8f7-f695f975c493)

**Comment by gedgon** [comment]
> How about we use `K`, `M`, `G`, etc. instead of the current `KB`, `MB`, `GB`, which will match the behavior of `ls -lah`, and it's also shorter, like this:


`ls` also displays prefixes incorrectly. Should be Ki, Mi there, but since we know we are counting bytes, KiB, MiB would be better. K, M would be ok with `ls -lah --si`

**Comment by gedgon** [comment]
Actually, I was wrong too. `ls` displays SI prefixes when uses power of 10 `--si`, so it returns the "correct" prefixes there (the difference with JEDEC is the lowercase `k` for kilo). Another issue is whether they should continue to use SI prefixes.

**Comment by sxyazi** [comment]
Hmm I'm a bit confused.

If I understand correctly, `kB` is power of 10, `KB` is power of 2, and `KB` is equivalent to `KiB`. So, Yazi's units are correct right?

**Comment by gedgon** [comment]
My apologies, I got completely messed up. Note for myself. Do not open bug reports in the middle of the night ;)

![20240126_04h52m03s_grim](https://github.com/sxyazi/yazi/assets/7478265/504a0bad-f735-442c-bffc-fd5e49c0b1f0)

**Comment by sxyazi** [comment]
Thank you for the explanation, I understand now.

However, `KiB` is just too long, and it's a challenge for limited space. I really don't want to explain to users every time that I'm using the correct unit.

So I've decided to switch to `K`, thus I only need to tell users that it behaves the same as `ls` and then just enjoy life. 🤦🏻

**Comment by sxyazi** [comment]
Done by https://github.com/sxyazi/yazi/commit/c02d69590bda49babe0883ff78341d92e557f381

**Comment by gedgon** [comment]
Once again, I apologize for the mix-up I made. I have edited the original report so it makes a little more sense.

I'm afraid https://github.com/sxyazi/yazi/commit/c02d69590bda49babe0883ff78341d92e557f381 doesn't change much. It follows the `ls` but not the standards for binary multiples and will continue to be a source of confusion. 

Unless I overlooked something (again), the only correct solution would be to use IEC prefixes.
